### PR TITLE
Can main function read fix 41

### PR DESCRIPTION
**Pull request by Muhammed-ELGendy**
Applied [ECUC_Can_00317] CanRxProcessing Enables / disables API Can_MainFunction_Read() for handling PDU reception events in polling mode.

### CODE REVIEW

**Comment by ahmedwahdan** [comment]
The PduInfoPtr is not declared. The code is not compiling

**Comment by Muhammed-ELGendy** [comment]
@ahmedwahdan 
This branch is left for more changes which would be made by @smsm2014

File: Software/bsw/static/Mcal/CAN/src/Can.c [hunk 1]
```diff
@@ -866,14 +877,25 @@ void Can_MainFunction_Read(void) {
                     Mailbox.ControllerId = controllerId;
                     // 2. inform CanIf using API below.
                     //TODO fill  PduInfoPt and protoype of can iF
-                  //  CanIf_RxIndication(Mailbox, PduInfoPtr);//We need to ask how to access these vars
+                    CanIf_RxIndication(Mailbox, PduInfoPtr);//We need to ask how to access these vars
```
Comment (ahmedwahdan) [inline Software/bsw/static/Mcal/CAN/src/Can.c]: PduInfoPtr not defined

Comment (smsm2014) [reply Software/bsw/static/Mcal/CAN/src/Can.c]: Fixed. Please see the new patch.

**Review by smsm2014 — APPROVED** [review]
Must be reviewed also by MahmoudNagah

File: Software/bsw/static/Mcal/CAN/src/Can.c [hunk 2]
```diff
@@ -914,19 +925,18 @@ uint8 controllerId; /*variable to count controllers number*/
 #if (CanDevErrorDetect == STD_ON)
         if (ControllerState[controllerId] == CAN_CS_UNINIT)
```
Comment (ahmedwahdan) [inline Software/bsw/static/Mcal/CAN/src/Can.c:913]: Can controllerId is not initilaized, we should loop over the configured CAN controllers

**Review by ahmedwahdan — CHANGES_REQUESTED** [review]
Please review the review comments

**Comment by ahmedwahdan** [comment]
@smsm2014  Please merge down from the DevCAN to get the changes, and resolve the review points.

File: Software/bsw/static/Mcal/CAN/src/Can.c (hunk 2 above)
Comment (smsm2014) [reply Software/bsw/static/Mcal/CAN/src/Can.c:913]: Sorry, but I didn't get your comment.
What you need here to be changed?

**Review by smsm2014 — APPROVED** [review]

**Review by ahmedwahdan — APPROVED** [review]

**Pull request merged**